scripts/install-linux.sh

```
@@ -0,0 +1,253 @@
+if [ -f "$SCRIPT_DIR/lib/common.sh" ]; then
+    source "$SCRIPT_DIR/lib/common.sh"
+    source "$SCRIPT_DIR/lib/github-api.sh"
+else
+    # If running from curl, download the library
+    echo "Downloading installation libraries..." >&2
+    TEMP_LIB_DIR=$(mktemp -d)
+    curl -fsSL "https://raw.githubusercontent.com/$REPO/main/scripts/lib/common.sh" -o "$TEMP_LIB_DIR/common.sh"
+    curl -fsSL "https://raw.githubusercontent.com/$REPO/main/scripts/lib/github-api.sh" -o "$TEMP_LIB_DIR/github-api.sh"
+    source "$TEMP_LIB_DIR/common.sh"
+    source "$TEMP_LIB_DIR/github-api.sh"
+    trap "rm -rf '$TEMP_LIB_DIR'" EXIT
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)